crates/uv-pep508/src/verbatim_url.rs

```diff
@@ -57,8 +57,51 @@ impl VerbatimUrl {
     }
 
     /// Parse a URL from a string.
-    pub fn parse_url(given: impl AsRef<str>) -> Result<Self, ParseError> {
-        let url = Url::parse(given.as_ref())?;
+    pub fn parse_url(given: impl AsRef<str>) -> Result<Self, VerbatimUrlError> {
```

Comment:
Should we share this code with `DisplaySafeUrl`?

Reply:
Sorry, I somehow missed this review earlier. Yeah, sharing this logic between the two makes a lot of sense to me.

(I see `VerbatimUrl` wraps `DisplaySafeUrl`, so perhaps it makes sense to push this check up to that layer?)